src/main/java/uk/gov/justice/digital/client/glue/GlueSchemaClient.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.justice.digital.client.glue;
+
+import com.amazonaws.services.glue.AWSGlue;
+import com.amazonaws.services.glue.model.EntityNotFoundException;
+import com.amazonaws.services.glue.model.GetSchemaVersionRequest;
+import com.amazonaws.services.glue.model.SchemaId;
+import com.amazonaws.services.glue.model.SchemaVersionNumber;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import lombok.val;
+import uk.gov.justice.digital.config.JobArguments;
+
+import java.util.Optional;
+
+@Singleton
+public class GlueSchemaClient {
+
+    private final AWSGlue glueClient;
+    private final String contractRegistryName;
+
+    @Inject
+    public GlueSchemaClient(GlueClientProvider glueClientProvider,
+                            JobArguments jobArguments) {
+        this.glueClient = glueClientProvider.getClient();
+        this.contractRegistryName = jobArguments.getContractRegistryName();
+    }
+
+    public Optional<String> getSchema(String schemaName) {
+        try {
+            val result = glueClient.getSchemaVersion(createRequestForSchemaName(schemaName));
+            val schemaData = result.getSchemaDefinition();
+            return Optional.of(schemaData);
+        }
+        catch (EntityNotFoundException e) {
+            return Optional.empty();
```

Comment:
no logger inside catch block

Reply:
logger inside the catch block is needed. It has lot of meaning what the exception is for debugging purpose

Reply:
Not necessarily. The caller should decide whether it's an error or not. In this case, the source reference service will fallback to the static resources that we have now and log. Later when we remove the fallbacks, it will throw an exception.